components/lib/calendar/Calendar.js

```diff
@@ -118,7 +118,7 @@ export const Calendar = React.memo(
                 }
             } catch (err) {
                 //invalid date
-                const value = props.keepInvalid ? rawValue : null;
+                const value = props.keepInvalid ? event.value : null;
```

Comment:
I actually think the line need to be this...

```js
const value = props.keepInvalid ? null : rawValue;
```

if you try your with `keepInvalid` it works but if you remove `keepInvalid` its allowing invalid characters

Reply:
The difference between keepInvalid true and false should only be on onBlur. 
You can input invalid characters with it true and false, but, when inBlur state value remains on keepInvalid = true, and on keepInvalid = false makes the state value null.

Reply:
Just try your fix above with both `keepInvalid={true}` and `keepInvalid={false}` you will see your fix does NOT work.  Then try mine you will see it does work.

Reply:
I did a couple of digging and found out why it didn't update. updateInputField didn't work properly.